### PR TITLE
Add a division visitor method to SolveExpression.

### PR DESCRIPTION
**Pull request by jingpu**
Moved #3202 to the master branch, as it is not mainly about the simplification rules.
The new simplification rule is already implemented in the new simplifier, so I will just create a PR to sync the simplifier test over.

### CODE REVIEW

File: src/Solve.cpp [hunk 1]
```diff
@@ -348,6 +348,61 @@ class SolveExpression : public IRMutator2 {
...
+        failed = old_failed || a_failed || b_failed;
+
+        const Add *add_a = a.as<Add>();
+        const Sub *sub_a = a.as<Sub>();
+        const Mul *mul_a = a.as<Mul>();
+
+        Expr expr;
+        if (a_uses_var && !b_uses_var) {
+            if (add_a && !a_failed &&
+                can_prove(add_a->a / b * b == add_a->a)) {
```
Comment (jingpu) [inline src/Solve.cpp]: Interesting. The correctness_likely test will time out inside this `can_prove` call. I am not sure how to fix. BTW, on the new simplifier branch, this code work (Woo, good).

**Comment by steven-johnson** [comment]
This PR has been sitting for a while; what's the status?

**Comment by jingpu** [comment]
There is probably a infinite recursion because the `correctness_likely` test times out. I will try debugging it. If it is not successful, I will probably wait until the new simplifier PR lands with which the test timeout goes away.

**Comment by jingpu** [comment]
Buildbot is happy after picking up the magnificent new simplifier change. PTAL.

**Pull request merged**